### ISSUE TITLE
TypeError when trying to parse request with int data

### ISSUE THREAD

**Issue by Grazfather** [flask-restful/flask-restful#598]
If the `Content-Type` header is "application/json" but the `data` is a _number_ then it raises a TypeError instead of a `BadRequest`

```python
        from flask.ext.restful import reqparse, inputs
        from flask import request
        import flask
        app = flask.Flask(__name__)
        with app.test_request_context(method="post", data=b"123", content_type="application/json"):
            parser = reqparse.RequestParser()
            parser.parse_args()
```

Change the `data` from `b"123"` to `b"abc"` and you get the expected error

**Comment by Grazfather** [comment]
```python
ERROR: test_bla (test_app.AppTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/grazfather/code/app/tests/test_app.py", line 57, in test_bla
    parser.parse_args()
  File "/Users/grazfather/code/app/venv/lib/python3.5/site-packages/flask_restful/reqparse.py", line 253, in parse_args
    req.unparsed_arguments = dict(Argument('').source(req))
  File "/Users/grazfather/code/app/venv/lib/python3.5/site-packages/flask_restful/reqparse.py", line 101, in source
    values.update(value)
  File "/Users/grazfather/code/app/venv/lib/python3.5/site-packages/werkzeug/datastructures.py", line 587, in update
    for key, value in iter_multi_items(other_dict):
  File "/Users/grazfather/code/app/venv/lib/python3.5/site-packages/werkzeug/datastructures.py", line 46, in iter_multi_items
    for item in mapping:
TypeError: 'int' object is not iterable

----------------------------------------------------------------------```

**Comment by Grazfather** [comment]
Was originally file in pallets/flask#1929

**Comment by tdpsk** [comment]
Thanks for the traceback. However the code you provided doesn't cause an error when I run it. Can you please provide the full code sample you are trying to run as well as the version of Python, flask and flask-restful you are using?

Also on a side note: using flask.ext.restful is deprecated and should be replaced by 

`from flask_restful import ...`

**Comment by Grazfather** [comment]
```bash
(venv) ~/code/app 11:14:54 (master)$ pip list|grep Flask
Flask (0.11.1)
Flask-RESTful (0.3.1)
(venv) ~/code/app 11:15:00 (master)$ python -V
```

```python
import unittest
from flask.ext.restful import reqparse, inputs
from flask import request
import flask

class AppTest(unittest.TestCase):
    def test_bla(self):
        app = flask.Flask(__name__)
        with app.test_request_context(method="post", data=b"123", content_type="application/json"):
            parser = reqparse.RequestParser()
            parser.parse_args()
```

**Comment by Grazfather** [comment]
Seems fixed in 0.3.5 :)

```bash
$ nosetests tests/test_app.py  -s
/Users/graziano/code/amazonappstore/venv/lib/python3.5/site-packages/flask/exthook.py:71: ExtDeprecationWarning: Importing flask.ext.restful is deprecated, use flask_restful instead.
  .format(x=modname), ExtDeprecationWarning
.
----------------------------------------------------------------------
Ran 1 test in 0.006s

OK
```
